src/planwise/client/current_project/handlers.cljs

```diff
@@ -48,20 +48,45 @@
     :transport :facilities-with-demand))
 
 (defn- project-loaded [db project-data]
-  (dispatch [:current-project/fetch-facility-types])
-  (dispatch [:regions/load-regions-with-geo [(:region-id project-data)]])
-  (db/new-viewmodel project-data))
+  (let [new-db (db/new-viewmodel project-data)]
+    (dispatch [:current-project/fetch-facility-types])
+    (dispatch [:regions/load-regions-with-geo [(:region-id project-data)]])
+    (if (:is-new? project-data)
+      (assoc-in new-db [:wizard :set] true)
+      new-db)))
+
+(defn- update-tabs-state [db next-tab]
+  (let [tabs (get-in db [:wizard :tabs])
+        new-tabs (into {}
+                       (map (fn [[tab state]]
+                              (if (and (= state :visiting) (not= tab next-tab))
+                                [tab :visited]
+                                [tab state])) tabs))]
+    (assoc-in db [:wizard :tabs] new-tabs)))
+
+(defn- update-next-tab-state [db next-tab]
+  (if (= :unvisited (get-in db [:wizard :tabs next-tab]))
+    (assoc-in db [:wizard :tabs next-tab] :visiting)
+    db))
+
+(defn- update-wizard-state [db next-tab]
```

Comment:
This should be named `visit-tab` or something like it and the logic can be simplified.